wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -799,7 +799,7 @@ private Subject sessionUser() {
 
     private String sessionUserNameOr(String nameIfNoUser) {
         final Subject subject;
-        if (EnvironmentContext.getCurrent() != null && (subject = EnvironmentContext.getCurrent().getSubject()) != null) {
+        if ((subject = EnvironmentContext.getCurrent().getSubject()) != Subject.ANONYMOUS) {
```

Comment:
@benoitf 
When can comparing by `==` work incorrect? Looks like only in case changing Subject from interface to class.
Equals can works incorrect if we change anonymous implementation to SubjectImpl("0000-000-00", "Anonynous", null, false) then equals won't show that it is really anonymous that are returned by EnvironmentContext, or it is some Subject with id "0000-000-00" and name "Anonynous". (I agree that it is crazy example)
Do you think that using equals there is correct variant?